### PR TITLE
add help cmd desc, closes revault#141

### LINKED ISSUE #141: Add a `help` command
Add a `help` command which prints all the available commands, so I don't have to open `doc/API.md` every time :D

### PR DESCRIPTION
**Pull request by vaibhavgarg237**
Fixes #141

### CODE REVIEW

**Comment by darosior** [comment]
Edited the OP to add a Github link which automatically closes the issue on this PR merge

File: doc/API.md [hunk 1]
```diff
@@ -29,6 +30,21 @@ Note that all addresses are bech32-encoded *version 0* native Segwit `scriptPubK
...
+
+#### Response
+
+| Field         | Type          | 
+| ------------- | ------------- | 
+| `name`        | string        | 
+| `parameters`  | string array  | 
+| `description` | string        | 
+
+Optional Parameters are enclosed in `[` `]`
```
Comment (darosior) [inline doc/API.md:45]: If i remember correctly this isn't the format of the response? It is something like:
```
"commands": [
    {},
    {}
]
```

Therefore, an array of objects. To follow what we use elsewhere you can say:

#### Response


| Field               | Type          | Description |
| ------------------ | ------------- |  ---------------- |
| "commands" | array        | Array of [help resource](help_resource) |

#### Help resource

| Field         | Type          | Description |
| ------------- | ------------- |  ---------------- |
| `name`        | string        |  The name of the command |
| `parameters`  | array  | The command parameters as strings, optional ones are enclosed in brackets (ie `[optional argument]`) |
| `description` | string        |  A description of the command |

**Review by darosior — COMMENTED** [review]
Thanks for taking care of that!

File: doc/API.md (hunk 1 above)
Comment (vaibhavgarg237) [reply doc/API.md:45]: Yeah, this looks perfect

Comment (vaibhavgarg237) [reply doc/API.md]: Pushed the required changes

File: doc/API.md [hunk 2]
```diff
@@ -29,6 +30,25 @@ Note that all addresses are bech32-encoded *version 0* native Segwit `scriptPubK
...
 
+### `help`
+
+Display all available commands.
+
+#### Response
+
+| Field      | Type  | Description                                 |
+| ---------- | ----- | ------------------------------------------- |
+| "commands" | array | Array of [help resource](#Help_resource)    |
```
Comment (darosior) [inline doc/API.md:41]: You should use backquotes to have `commands` be rendered as pre-formatted. Also the link is wrong (it was copy pasted from my example which was just an instance. I tested and it should be `(#help-resource)`.

**Review by darosior — COMMENTED** [review]
One final comment and we are good to go!

File: doc/API.md (hunk 2 above)
Comment (vaibhavgarg237) [reply doc/API.md]: Yeah, I missed that!
Made the required changes

**Comment by darosior** [comment]
utACK 0b84d8a -- Thanks for this and for following-up with my nits!

**Comment by vaibhavgarg237** [comment]
> utACK [0b84d8a](https://github.com/revault/revaultd/commit/0b84d8a433f1128acc9f26e44b69e808c057cc1b) -- Thanks for this and for following-up with my nits!
It's my pleasure. Learned a lot about these nits. This won't happen again from my side.

**Pull request merged**